db/models.js

```diff
@@ -1,8 +1,26 @@
 const db = require("./connection");
 
 exports.fetchTopics = () => {
-  return db.query("SELECT * FROM topics").then(({ rows }) => {
-    console.log(rows);
+  return db.query("SELECT * FROM topics;").then(({ rows }) => {
     return rows;
   });
 };
+
+exports.fetchArticleById = (articleId) => {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id=$1;`, [articleId])
+
+    .then(({ rows }) => {
+      const article = rows[0];
+      if (!article) {
+        return Promise.reject({
+          status: 404,
+          msg: `No article found with article ID: ${articleId}`,
+        });
+      }
+      return article;
+    })
+    .catch((err) => {
+      return Promise.reject(err);
```

Comment:
this catch with a Promise.reject is unnecessary - if there is an error thrown it will be thrown down the promise chain so you should see it caught in the catch of your controller without the need for this catch in your model

Reply:
Ok that makes sense!